api/endpoints/upload.py

```
@@ -14,11 +15,39 @@ def upload_document(file: UploadFile = File(...)):  # noqa: B008
-        return UploadResponse(filename=file.filename, message="File uploaded successfully.")
+        try:
+            saved_path = save_upload_file(file.file, file.filename)
+        except ValueError as ve:
+            raise HTTPException(status_code=400, detail=str(ve)) from ve
+        parsing_status = "success"
+        text_preview = None
+        if file.content_type == "application/pdf":
+            try:
+                text = extract_text_from_pdf(saved_path)
+                text_preview = text[:500] if text else None
+            except Exception:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)